mvcc/kvstore.go

```diff
@@ -145,14 +145,14 @@ func NewStore(lg *zap.Logger, b backend.Backend, le lease.Lessor, ci cindex.Cons
 
 func (s *store) compactBarrier(ctx context.Context, ch chan struct{}) {
 	if ctx == nil || ctx.Err() != nil {
-		s.mu.Lock()
 		select {
 		case <-s.stopc:
 		default:
+			s.mu.Lock()
```

Comment:
Can we document this why this is safe? (e.g. `s.stopc` is only updated in restore operation, which is called by apply snapshot call, compaction and apply snapshot requests are serialized by raft, and do not happen at the same time)

Reply:
thanks.updated. I understand the mutex here just to protect s.fifoSched.